tip-wlan/charts/opensync-gw-cloud/templates/deployment.yaml

```diff
@@ -28,139 +28,145 @@ spec:
-        - name: {{ include "common.name" . }}-readiness-prov
-          image: alpine
-          imagePullPolicy: {{ .Values.global.pullPolicy }}
-          command:
-          - sh
-          - -c
-          - |
-            if [ {{ $cloudeployment }} = false ]
-            then          
-              echo "151.101.112.249 dl-cdn.alpinelinux.org" >> /etc/hosts
-              echo "Added name-resolution for local deployments"
-            fi
-            apk add curl
-            url=https://{{ $prov }}/ping
-            counter=0
-            status=$(curl --insecure --head --location --connect-timeout 5 --write-out %{http_code} --silent --output /dev/null ${url});
-            while [ $counter -lt 10 ] && [ $status -ne 200 ] 
-            do
-              echo ${url} service isnt ready. Tried $counter times
-              sleep 5
-              counter=`expr $counter + 1`
-              status=$(curl --insecure --head --location --connect-timeout 5 --write-out %{http_code} --silent --output /dev/null ${url});
-              echo Http Response code of ping request = $status
-            done
-            if [ $status -eq 200 ]
-            then
-              echo ${url} service is ready!
-            else
-              echo ${url} service failed to respond after 50 secs
-              exit 1
-            fi
-        - name: {{ include "common.name" . }}-readiness-ssc
-          image: alpine
-          imagePullPolicy: {{ .Values.global.pullPolicy }}
-          command:
-          - sh
-          - -c
-          - |
-            if [ {{ $cloudeployment }} = false ]
-            then          
-              echo "151.101.112.249 dl-cdn.alpinelinux.org" >> /etc/hosts
-              echo "Added name-resolution for local deployments"
-            fi
-            apk add curl
-            url=https://{{ $ssc }}/ping
-            counter=0
-            status=$(curl --insecure --head --location --connect-timeout 5 --write-out %{http_code} --silent --output /dev/null ${url});
-            while [ $counter -lt 10 ] && [ $status -ne 200 ] 
-            do
-              echo ${url} service isnt ready. Tried $counter times
-              sleep 5
-              counter=`expr $counter + 1`
-              status=$(curl --insecure --head --location --connect-timeout 5 --write-out %{http_code} --silent --output /dev/null ${url});
-              echo Http Response code of ping request = $status
-            done
-            if [ $status -eq 200 ]
-            then
-              echo ${url} service is ready!
-            else
-              echo ${url} service failed to respond after 50 secs
-              exit 1
-            fi        
-      {{- end }}
+      - name: wait-for-services
+        image: opsfleet/depends-on
+        args:
+        - "-service=tip-wlan-opensync-mqtt-broker" # fixme
+        - "-service=tip-wlan-wlan-prov-service" # fixme
+        - -check_interval=5
+      #   - name: {{ include "common.name" . }}-mqtt-readiness
```

Comment:
@eugenetaranov if you're sure the commented out code isn't needed anymore, let's delete it entirely instead of commenting out. We have git history if we would like to revert back or check what was deleted in the future.

Reply:
@eugenetaranov : I am not sure what's the way to query a service readiness in the `opsfleet/depends-on` image but we need to make sure that prov, ssc and mqtt services are running and responding to requests (/ping for prov and ssc and 'Aliveness check' for Mqtt) before we start Opensync-Gw svc.
Also when running in Integrated env (prov, ssc and spc as one process), we need to make sure that the IntegratedComponent responds to /ping .

Reply:
> what's the way to query a service readiness in the opsfleet/depends-on image
it does not support this, it relies on pod passing k8s healthchecks

> we need to make sure that prov, ssc and mqtt services are running and responding to requests 
this is a task for startup/liveness/readiness probes to make sure the service is alive

having liveness probes defined in 2 places does not improve application health but makes it harder to maintain and slowdowns startup

what do you think @rahulsharma2020 , @leonidlm , @cbuschfb ?

Reply:
I see. Thanks for the explanation @leonidlm, understand it now.
In that case, the /ping for `prov` and `ssc` processes are good. 
However for mqtt, it's different:
- the readiness probe currently checks for `tcpSocket` on a particular port. 
- Whereas for checking mqtt readiness from Opensync-gw pod, we use the mosquitto_pub (a mosquitto client - https://mosquitto.org/man/mosquitto_pub-1.html) to make sure it's alive. 
So to use the `depends-on` image for MQTT, we should change it's readiness probe to use mosquitto_pub instead.

Reply:
@rahulsharma2020 sounds like a good idea! Using mosquitto_pub instead of tcpSocket for readiness probe is probably a good idea anyway, unrelated to depends-on. @eugenetaranov WDYT?

Reply:
@leonidlm already added here https://github.com/Telecominfraproject/wlan-cloud-helm/blob/7fee3eca5d2673d7225840339394b1358c260772/tip-wlan/charts/opensync-mqtt-broker/templates/statefulset.yaml#L84 as startupProbe

Reply:
Thanks @eugenetaranov-opsfleet . Looks good. Should we just put that in readiness/liveliness probe instead of startupProbe or do you see any issue there?

Reply:
@rahulsharma2020 I think it's sufficient to have it checked only once at startup but if you feel like it makes sense to check it every minute or so let's do it. The only thing we need to make sure is that test messages auto expire and do not fill disk or cause other issues.